src/NServiceBus.Core/Reliability/SynchronizedStorage/CompletableSynchronizedStorageSessionExtensions.cs

```diff
@@ -14,75 +14,38 @@ public static class CompletableSynchronizedStorageSessionExtensions
             ContextBag contextBag)
         {
-            if (await session.TryOpen(outboxTransaction, contextBag).ConfigureAwait(false))
+            if (await sessionAdapter.TryOpen(outboxTransaction, contextBag).ConfigureAwait(false))
             {
                 return;
             }
 
-            if (await session.TryOpen(transportTransaction, contextBag).ConfigureAwait(false))
+            if (await sessionAdapter.TryOpen(transportTransaction, contextBag).ConfigureAwait(false))
             {
                 return;
             }
 
-            await session.Open(contextBag).ConfigureAwait(false);
+            await sessionAdapter.Open(contextBag).ConfigureAwait(false);
         }
-
-        /// <summary>
-        /// Tries to open the storage session with the provided outbox transaction.
-        /// </summary>
-        /// <param name="session">The storage session.</param>
-        /// <param name="transaction">Outbox transaction.</param>
-        /// <param name="context">Context.</param>
-        /// <returns><c>true</c> when the session was opened; otherwise <c>false</c>.</returns>
-        public static Task<bool> TryOpen(this CompletableSynchronizedStorageSession session,
-            OutboxTransaction transaction, ContextBag context) =>
-            ((CompletableSynchronizedStorageSessionAdapter)session).TryOpen(transaction, context);
-
-        /// <summary>
-        /// Tries to open the storage session with the provided transport transaction.
-        /// </summary>
-        /// <param name="session">The storage session.</param>
-        /// <param name="transportTransaction">Transport transaction.</param>
-        /// <param name="context">Context.</param>
-        /// <returns><c>true</c> when the session was opened; otherwise <c>false</c>.</returns>
-        public static Task<bool> TryOpen(this CompletableSynchronizedStorageSession session,
-            TransportTransaction transportTransaction, ContextBag context) =>
-            ((CompletableSynchronizedStorageSessionAdapter)session).TryOpen(transportTransaction, context);
-
-        /// <summary>
-        /// Opens the storage session.
-        /// </summary>
-        /// <param name="session">The storage session.</param>
-        /// <param name="context">The context information.</param>
-        public static Task Open(this CompletableSynchronizedStorageSession session, ContextBag context) =>
-            ((CompletableSynchronizedStorageSessionAdapter)session).Open(context);
-
-        /// <summary>
-        /// Returns the underlying, persistence specific, storage session.
-        /// </summary>
-        /// <param name="session">The storage session.</param>
-        public static CompletableSynchronizedStorageSession GetAdaptedSession(this CompletableSynchronizedStorageSession session) =>
-            ((CompletableSynchronizedStorageSessionAdapter)session).AdaptedSession;
```

Comment:
@timbussmann We discussed this during this morning's sync. I agree this looks very confusing, but keeping v7 and v8 aligned should be a priority, especially due to how we work and how much context is lost around these changes.

Would it be helpful if we modify this as proposed in https://github.com/Particular/NServiceBus/pull/6529 ?